langchain/src/schema/runnable/remote.ts

```diff
@@ -0,0 +1,312 @@
+      return new ChatPromptValue({
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        messages: obj.messages.map((msg: any) => revive(msg)),
+      });
+    }
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const innerRevive: (key: string) => [string, any] = (key: string) => [
+      key,
+      revive(obj[key]),
+    ];
+    const rtn = Object.fromEntries(keysArr.map(innerRevive));
+    return rtn;
+  }
+  return obj;
+}
+function deserialize<RunOutput>(str: string): RunOutput {
+  const obj = JSON.parse(str);
+  return revive(obj);
+}
+
+function withoutCallbacks(
+  options?: RunnableConfig
+): Omit<RunnableConfig, "callbacks"> {
+  const rest = { ...options };
+  delete rest.callbacks;
+  return rest;
+}
+
+export class RemoteRunnable<
+  RunInput,
+  RunOutput,
+  CallOptions extends RunnableConfig
+> extends Runnable<RunInput, RunOutput, CallOptions> {
+  private url: string;
+
+  private options?: RemoteRunnableOptions;
+
+  lc_namespace = ["langchain", "schema", "runnable", "remote"];
+
+  constructor(url: string, options?: RemoteRunnableOptions) {
+    super();
+    this.url = url.replace(/\/$/, ""); // remove trailing slash
+    this.options = options;
+  }
+
+  private async post<Body>(path: string, body: Body) {
+    return await fetch(`${this.url}${path}`, {
+      method: "POST",
+      body: JSON.stringify(body),
+      headers: {
+        "Content-Type": "application/json",
+      },
+      signal: AbortSignal.timeout(this.options?.timeout ?? 5000),
+    });
+  }
+
+  async invoke(input: RunInput, options?: CallOptions): Promise<RunOutput> {
+    const [config, kwargs] =
+      this._separateRunnableConfigFromCallOptions(options);
+    const response = await this.post<{
+      input: RunInput;
+      config: RunnableConfig;
+      kwargs: Omit<Partial<CallOptions>, keyof BaseCallbackConfig>;
+    }>("/invoke", {
+      input,
+      config: withoutCallbacks(config),
+      kwargs: kwargs ?? {},
+    });
+    return revive((await response.json()).output) as RunOutput;
+  }
+
+  async _batch(
+    inputs: RunInput[],
+    options?: CallOptions[],
+    _?: (CallbackManagerForChainRun | undefined)[],
+    batchOptions?: RunnableBatchOptions
+  ): Promise<(RunOutput | Error)[]> {
+    if (batchOptions?.returnExceptions) {
+      throw new Error("returnExceptions is not supported for remote clients");
+    }
+    const arr = options?.map((opts) =>
+      this._separateRunnableConfigFromCallOptions(opts)
+    );
+    const tup = arr?.reduce(
+      ([pc, pk], [c, k]) =>
+        [
+          [...pc, c],
+          [...pk, k],
+        ] as [
+          RunnableConfig[],
+          Omit<Partial<CallOptions>, keyof BaseCallbackConfig>[]
+        ],
+      [[], []] as [
+        RunnableConfig[],
+        Omit<Partial<CallOptions>, keyof BaseCallbackConfig>[]
+      ]
+    );
+    const [configs, kwargs] = tup ?? [undefined, undefined];
```

Comment:
I think these should default to empty objects?

Reply:
Will bring this up in slack. Even this is a slightly different implementation than python, which just passes a "whole" kwargs object in with the `batch` call, rather than assuming a different object per request in the batch. I actually like the js version better, but should make sure we're ok with that in general.